Terminal.Gui/Core/Application.cs

```diff
@@ -277,6 +277,14 @@ static void OnQuitKeyChanged (KeyChangedEventArgs e)
 		/// </remarks>
 		public static Action Iteration;
 
+		/// <summary>
+		///   This event is raised on each iteration of the <see cref="MainLoop"/> after all processes are completed.
+		/// </summary>
+		/// <remarks>
+		///   See also <see cref="Timeout"/>
+		/// </remarks>
+		public static Action IterationComplete;
```

Comment:
Nothing wrong with this as a new primitive event, but would it be better for this PR if there were an event that was draw related?

In v2, i'd like us to revisit how `Redraw` and the related events are structured. I started this in my View2 PR...

- `Redraw` is mis-named. It should just be `Draw` (it also shouldn't take a `Bounds` because implementors can just use `this.Bounds`; this is a cause of a bunch of bugs BTW).
- `Draw` should simply call
  - `OnDraw`
  - `OnDrawFrames`
  - `OnDrawContent`
  - `OnDrawContentComplete`

Each of those invokes an event and each is cancellable (except OnDrawContentComplete).

With this, your tests could subscribe to `DrawContentComplete` instead of `IterationComplete`?

Reply:
That the point @tig I mentioned in this comment https://github.com/gui-cs/Terminal.Gui/pull/2479#issuecomment-1493371057. It's almost have this ready, only finishing some unit tests. The `IterationComplete` will be removed. Only the `DrawContentComplete` is more accurate to draw anything in the current run state toplevel, because some toplevels consume the `Redraw` method without calling the `OnDrawContentComplete` method later,